lib/puppet/provider/package/pkgng.rb

```diff
@@ -61,7 +61,7 @@ def self.instances
   end
 
   def self.prefetch(resources)
-    packages = instances
+    packages = instances || []
```

Comment:
@Iristyle I believe I've adjusted the code to be in line with your desires.  It feels cleaner to handle the exception in the instances.  I've left the old test in place and added a new one as well.  I've also updated the commit message and the PR message.  The ticket wording should be accurate as is.  Let me know how this looks now.